solidity/test/bridge/Bridge.test.ts

```diff
@@ -968,1024 +1023,1171 @@ describe("Bridge", () => {
+        expect(mainUtxoHash).to.be.equal(expectedMainUtxo)
+      })
 
-        after(async () => {
-          await restoreSnapshot()
-        })
+      it("should update the depositor's balance", async () => {
+        // The sum of sweep tx inputs is 20000 satoshi. The output
+        // value is 18500 so the transaction fee is 1500. There is
+        // only one deposit so it incurs the entire transaction fee.
+        // The deposit should also incur the treasury fee whose
+        // initial value is 0.05% of the deposited amount so the
+        // final depositor balance should be cut by 10 satoshi.
+        expect(
+          await bank.balanceOf(data.deposits[0].reveal.depositor)
+        ).to.be.equal(18490)
+      })
 
-        it("should revert", async () => {
-          // Each header encodes a `difficulty target` field in bytes 72-76.
-          // The given header's hash (interpreted as uint) must be bigger than
-          // the `difficulty target`. To test this scenario, we change the
-          // last byte of the last header in such a way their hash becomes
-          // lower than their `difficulty target`.
-          const properHeaders = data.sweepProof.bitcoinHeaders.toString()
-          data.sweepProof.bitcoinHeaders = `${properHeaders.substring(
-            0,
-            properHeaders.length - 2
-          )}ff`
+      it("should transfer collected treasury fee", async () => {
+        expect(await bank.balanceOf(treasury.address)).to.be.equal(10)
+      })
 
-          await expect(runSweepScenario(data)).to.be.revertedWith(
-            "Insufficient work in a header"
-          )
-        })
+      it("should emit DepositsSwept event", async () => {
+        await expect(tx)
+          .to.emit(bridge, "DepositsSwept")
+          .withArgs(reveal.walletPubKeyHash, data.sweepTx.hash)
       })
+    })
 
-      context(
-        "when accumulated difficulty in headers chain is insufficient",
-        () => {
-          let otherBridge: Bridge
-          const data: SweepTestData = JSON.parse(
-            JSON.stringify(SingleP2SHDeposit)
-          )
-          // Take wallet public key hash from first deposit. All
-          // deposits in same sweep batch should have the same value
-          // of that field.
-          const { walletPubKeyHash } = data.deposits[0].reveal
+    context("when the wallet state is neither Live or MovingFunds", () => {
+      const data: SweepTestData = SingleP2SHDeposit
+      const { fundingTx, reveal } = data.deposits[0]
+
+      const testData = [
+        {
+          testName: "when wallet state is Unknown",
+          walletState: walletState.Unknown,
+        },
+        {
+          testName: "when wallet state is Closed",
+          walletState: walletState.Closed,
+        },
+        {
+          testName: "when wallet state is Terminated",
+          walletState: walletState.Terminated,
+        },
+        // TODO: Implement tests for state `Closed` when it's added
```

Comment:
This is about `Closing` state, not `Closed`. However, I think we can drop that TODO anyway. If we'll introduce the `Closing` state, we will be forced to revisit all test cases like that anyway.

Reply:
Removed the TODO in b90c534b4001e7a99cce681451174b1eee74f5fc.